test/langtools/tools/javac/sealed/ValidateJarWithSealedAndRecord.java

```diff
@@ -60,7 +51,7 @@ void writeFile(String f, String contents) throws IOException {
     }
 
     void generateFilesNeeded() throws Exception {
-        sun.tools.jar.Main jarGenerator = new sun.tools.jar.Main(System.out, System.err, "jar");
+        ToolProvider jarTool = ToolProvider.findFirst("jar").orElseThrow(() -> new RuntimeException("jar tool not found"));
```

Comment:
I would suggest doing something like the following which we do in other tests:

```
    // Use the ToolProvider interface for accessing the jar tool
    private static final ToolProvider JAR_TOOL = ToolProvider.findFirst("jar")
            .orElseThrow(() -> new RuntimeException("jar tool not found")
            );
```
Then just use `JAR_TOOL.run(...)`  as it is easier to read

Reply:
I'm creating two instance of jarTool in the test that's why I think that it makes more sense to declare it as a local variable, definitely not a final one. I have to create two instances because the options of the first invocation are not cleared up and interfere with the options passed to the second invocation

Reply:
there are combinations of options that can't be set at the same time for the `jar` tool and the API doesn't allow to `clear` options, that's why the only way I found was to create a new instance of the `jar` tool. For example `--create` and `--validate` can't be used together, given that the first invocation of the `jar` tool uses `--create` and I have no way to clear the passed options that forces me to create a new instance of the `jar` tool

Reply:
It sounds like a bug in jar tool.